osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/config/AbstractPersistenceConfig.java

```diff
@@ -167,6 +181,26 @@ protected LocalContainerEntityManagerFactoryBean entityManagerFactory(final Stri
         return entityManagerFactoryBean;
     }
 
+    private void checkForDeprecatedHibernateNamingStrategyConfiguration() {
+        String deprecatedProperty = null;
+        try {
+            deprecatedProperty = this.environment
+                    .getRequiredProperty(PROPERTY_NAME_HIBERNATE_NAMING_STRATEGY_DEPRACATED);
+        } catch (final IllegalStateException e) {
+            // The property is expected to be absent.
+            LOGGER.debug("IllegalStateException", e);
+            return;
+        }
+
+        final String message = String.format(
+                "Using '%s=%s' is deprecated and no longer works with Hibernate 5.X! Use '%s=%s' instead!",
+                PROPERTY_NAME_HIBERNATE_NAMING_STRATEGY_DEPRACATED, deprecatedProperty,
+                PROPERTY_NAME_HIBERNATE_NAMING_STRATEGY, HibernateNamingStrategy.class.getName());
+
+        LOGGER.error(message);
+        throw new RuntimeException(message);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS00112)